DataDog/dd-trace-rb
# Revert profiling back to using the agent_settings generated in Datadog::Core::Configuration::Components.initalize

**What does this PR do?**
Reverts the profiling component back to using the `AgentSettings` generated in `Datadog::Core::Configuration::Components.initalize` rather than having its own `AgentSettingsResolver` and generating `AgentSettings` in the profiling component. Also, completely remove the `Profiling::AgentSettingsResolver`.

**Motivation:**
[The conversation](https://github.com/DataDog/dd-trace-rb/pull/3067/files#r1305381594) with @ivoanjo.

Profiling sees no use case for any custom behavior for `AgentSettingsResolver` for the foreseeable future. The goal of #3067 was to remove non-core dependencies from `Core::Configuration::AgentSettingsResolver` and this reversion doesn't negatively impact that goal.  

**Additional Notes:**


**How to test the change?**
No functional change should occur in this PR. The profiler should continue to connect to the agent normally. 

**For Datadog employees:**
credentials of any kind, I've requested a review from `@DataDog/security-design-and-guidance`.

Unsure? Have a question? Request a review!
